Cookbook/Proposals/SplitOctopus.md

```diff
@@ -0,0 +1,65 @@
+# Split Octopus codebase
+
+* Author(s): Giorgos Tsiapaliokas
+* Review Managers: David Rodrigues, Yasuhiro Inami, Olivier Halligon
+
+## Introduction
+
+From the early versions of the codebase Octopus was the place were features were added.
+Unfortunately Octopus hasn't adapted as the rest of the codebase has.
+In the iOS codebase we use frameworks in order to organize the code and Octopus isn't one.
+
+## Motivation
+
+While there is still the need for having a place for small features Octopus has become a massive container which contains both features and services.
+Also from a tech POV Octopus is just a folder which is being reused in different targets.
+I am sure we can do better than this :)
+
+## Proposed solution
+
+Split Octopus into two different frameworks.
+The proposed names are `BabylonFeatures` and `BabylonApp`.
```

Comment:
Can I propose to make ability to make other apps more independent?
To the extend that we would just have something like

```swift
final class AppDelegate {
   let app: App
   init() {
     app = BabylonApp()
   }

open func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
    app.run()
}
}
```

Current approach of feature switches is not perfect as there may be some features that are only available on specific app

Reply:
I see the benefits of having a new `AppDelegate` for each app.

This would remove some of the configuration values/tokens which are needed, but feature switches would still be required. Right?

Reply:
@sergdort how this is different than having its won app delegate implemented in each app? we still have common tasks to perform for all the apps

Reply:
The idea I head in mind is really separate app for each client so if they would have their own Flow they can implement them freely without worrying that it gonna break other apps. 
Essentially what other apps would use are common Frameworks + Their features. So we would have a separate project without adding files to the multiple targets like we do now

Reply:
Yes, that would be good. But we still will need to have common things to perform for each app as they are configured identically using the environment values. So it will be like extracting shared app delegate code to the framework but then we don't solve the problem of code sharing, or having this shared code duplicated which also have tradeoffs. So probably we would still have some kind of BaseApp 🤷‍♂ 
If what you are talking about is for example how these apps will configure let's say their tab bar, then I guess the architecture we could have is to have a tab bar provided by a framework and provide an interface for apps to configure it, rather than how we have now using feature switches in the tab bar. This way the configuration point will be in these app instances. That also intersects with feature switches proposal then.

Reply:
I still don't understand what `BabylonApp` framework will be. That naming seems too generic.
(Maybe the Google Document linked below explains that better, but since I don't have access and can't view it, all this seems pretty vague to me)